### PR TITLE
Add analyzer descriptions

### PR DESCRIPTION
**Pull request by ddmler**
Hey!

Type: 

Link to issue: #234 

This pull request affects the following components **(please check boxes):**

* [ ] Core
* [x] Analyzer
* [ ] Compiler
* [ ] Control Flow Graph
* [x] Documentation

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/ovr/phpsa/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [ ] I wrote some tests for this PR.

Small description of change:

Added descriptions to analyzers, min php version for finalstatic analyzer, updated documentation.
Would be cool if dump-reference command could output the documentation file so we only have to update the texts once.

### CODE REVIEW

File: src/Analyzer/Pass/Expression/FunctionCall/ArgumentUnpacking.php [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 class ArgumentUnpacking extends AbstractFunctionCallAnalyzer
 {
+    const DESCRIPTION = 'Checks for use of `func_get_args()` and suggests the use of argument unpacking. (... operator)';
```
Comment (K-Phoen) [inline src/Analyzer/Pass/Expression/FunctionCall/ArgumentUnpacking.php]: Shouldn't this analyzer require PHP >= 5.6?

**Comment by K-Phoen** [comment]
Great work! 👍 

Could you also add a commit to update the `.phpsa.yml` file?
You can re-generate-it using `./bin/phpsa config:dump-reference > .phpsa.yml`

**Comment by K-Phoen** [comment]
> Would be cool if dump-reference command could output the documentation file so we only have to update the texts once.

I don't think that this command should dump this documentation. But we definitely could provide another command to dot it :)

File: src/Analyzer/Pass/Expression/FunctionCall/ArgumentUnpacking.php (hunk 1 above)
Comment (ovr) [inline src/Analyzer/Pass/Expression/FunctionCall/ArgumentUnpacking.php]: Yes, I will push related changes @K-Phoen

**Comment by ovr** [comment]
Thanks @ddmler! 🐟

**Comment by ddmler** [comment]
@K-Phoen I got confused with the command. Then we need a new one :+1: 
I had added a short list which of them need a min php version in issue #234

**Pull request merged**